src/logic/utils.ts

```diff
@@ -1,3 +1,38 @@
+import { PLAYER_MINIMUM_GAMES_REQUIRED, PLAYER_MINIMUM_WINS_REQUIRED } from "../components/constants";
+import { Match } from "../types/domain/Match";
+import { Player } from "../types/domain/Player";
+
 export function getWinRatePercentage(winCount: number, totalCount: number) {
     return totalCount > 0 ? Math.round((winCount / totalCount) * 100) : 0;
 }
+
+export function avgWinTurn(player: Player) {
+    // Early exit conditions
+    // Don't show if player has fewer than 10 matches or 5 wins all time
+    if (player.wins < PLAYER_MINIMUM_WINS_REQUIRED || player.matches.length < PLAYER_MINIMUM_GAMES_REQUIRED) {
+        return "Insufficient data";
+    }
+
+    // Create win turns array
+    const winTurns = [];
+
+    // Get win turn for every match win
+    for (const match of player.matches) {
+        if (player.name === match.winner) {
+            // Exclude wins without turns data
+            if (Number(match.numberOfTurns) > 1) {

```

Comment:
no cEDH gameplay here? :D 

The joke being is that technically games can last 1 turn, and we should have a winner in those instances and not skip them.